dm/master/server.go

```diff
@@ -2027,3 +2323,384 @@ func (s *Server) workerArgsExtractor(args ...interface{}) (workerrpc.Client, str
 
 	return cli, worker, nil
 }
+
+// WatchRequest watches requests in etcd, and handle these request, and write response to etcd.
+func (s *Server) WatchRequest(ctx context.Context) {
+	watchCh := s.etcdClient.Watch(ctx, defaultOperatePath, -1)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case wresp, ok := <-watchCh:
+			if !ok {
+				log.L().Info("etcd's watch channel is closed", zap.String("watch key", defaultOperatePath))
+				return
+			}
+
+			if wresp.Err() != nil {
+				log.L().Warn("watch etcd failed", zap.Error(wresp.Err()))
+				continue
+			}
+
+			for _, ev := range wresp.Events {
+				// only need handle put event
+				if ev.Type != mvccpb.PUT {
+					continue
+				}
+
+				operate := &pb.Operate{}
+				err := operate.Unmarshal(ev.Kv.Value)
+				if err != nil {
+					log.L().Error("unmarshal operate failed", zap.String("event key", string(ev.Kv.Key)), zap.Int64("revision", wresp.Header.Revision), zap.Error(err))
+					continue
+				}
+
+				if operate.Stage == pb.OperateStage_Doing || operate.Stage == pb.OperateStage_Done {
+					// this request already handled, ignore it
+					continue
+				}
+
+				// only master leader need handle request
+				if s.election.IsLeader() {
+					go s.handleRequest(string(ev.Kv.Key), operate)
```

Comment:
should we manage these goroutines？

Reply:
I think it is unnecessary,  just abandon these requests when leader is exit. dmctl will get error, and user can  execute command again later.

Reply:
What I care about is correctness. If the leader migrates and the transaction processing has not ended, will there be unexpected consequences?